react-app/src/components/AppSideBar/AppSideBar.tsx

```diff
@@ -89,162 +88,150 @@ const AppSideBar: React.FC<AppSideBarProps> = (props) => {
+            />
+          ) : (
+            <div
+              className={cn(
+                "order-3",
+                "flex-col",
+                "gap-y-6",
+                "hidden",
+                "sm:flex"
+              )}
+            >
+              <Divider />
+              <UserInfoPanel
+                className={cn("order-3")}
+                userInfo={userInfo}
+                onClickSend={transaction.openSendTokenModal}
+                onClickReceive={transaction.openReceiveTokenModal}
+                onClickReward={transaction.openCollectRewardsModal}
+                onClickReinvest={onReinvest}
+              />
+            </div>
+          )}
           <div
             className={cn(
-              "flex-0",
-              "flex",
+              "hidden",
+              "sm:flex",
               "flex-col",
-              "px-3",
-              "pt-3",
-              "sm:p-0",
-              "sm:w-72"
+              "gap-y-4",
+              "mb-6",
+              wallet.status === ConnectionStatus.Connected
+                ? "order-2"
+                : "order-3"
             )}
           >
-            <div className={cn("flex", "flex-row", "order-1", "sm:flex-col")}>
-              <Header
-                chainHealth={
-                  isRequestStateLoaded(chainHealthRequestState)
-                    ? chainHealthRequestState.data
-                    : undefined
-                }
-              />
-              <IconButton
-                icon={isMenuActive ? IconType.X : IconType.Menu}
-                size={24}
-                className={cn("sm:hidden")}
-                onClick={toggleMobileMenuMenu}
-              />
-            </div>
-            {wallet.status !== ConnectionStatus.Connected ? (
-              <LoginPanel
-                className={cn("order-2", "my-6")}
-                onConnect={wallet.openConnectWalletModal}
+            {wallet.status === ConnectionStatus.Connected && (
+              <AddressBar
+                address={userInfo?.address ?? ""}
+                onDisconnect={wallet.disconnect}
               />
-            ) : (
-              <div
-                className={cn(
-                  "order-3",
-                  "flex-col",
-                  "gap-y-6",
-                  "hidden",
-                  "sm:flex"
-                )}
-              >
+            )}
+            <Divider />
+            <AppNavigationMenu
+              activeRoute={location.pathname}
+              className={cn("pb-6")}
+              onMenuItemSelect={closeMobileMenu}
+            />
+          </div>
+        </div>
+        <div
+          className={cn(
+            "flex-1",
+            "flex",
+            "relative",
+            "px-3",
+            "sm:max-w-screen-2xl",
+            "sm:px-0"
+          )}
+        >
+          <div
+            className={cn(
+              isMenuActive ? "flex" : "hidden",
+              "h-[calc(100vh_-_5.25rem)]",
```

Comment:
looks a bit magic 🤔 how did you come up with this? maybe leave a comment?

Reply:
I came up with this inspecting the height of the mobile header. 

I agree its a bit magic, it was left over from previous fix attempt when I try to fix the issue without refactoring the entire AppSidebar in #130. I believe this is removed in later commits.